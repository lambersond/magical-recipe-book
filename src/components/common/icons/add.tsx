import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function AddIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='AddIcon'
      {...props}
      d='M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z'
    />
  )
}
