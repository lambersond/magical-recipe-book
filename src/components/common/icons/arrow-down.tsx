import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function ArrowDownIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='ArrowDownIcon'
      {...props}
      d='M5.23 7.21a.75.75 0 011.06.02L10 10.94l3.71-3.71a.75.75 0 111.06 1.06l-4.24 4.24a.75.75 0 01-1.06 0L5.23 8.27a.75.75 0 01.02-1.06z'
    />
  )
}
