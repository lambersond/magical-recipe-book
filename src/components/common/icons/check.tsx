import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function CheckIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='CheckIcon'
      {...props}
      d='M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z'
    />
  )
}
