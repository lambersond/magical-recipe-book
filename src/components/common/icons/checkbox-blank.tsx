import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function CheckboxBlankIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='CheckboxBlankIcon'
      {...props}
      d='M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2'
    />
  )
}
