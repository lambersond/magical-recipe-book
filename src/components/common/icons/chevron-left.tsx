import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function ChevronLeftIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='ChevronLeftIcon'
      {...props}
      d='M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z'
    />
  )
}
