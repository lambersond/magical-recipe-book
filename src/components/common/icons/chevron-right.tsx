import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function ChevronRightIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='ChevronRightIcon'
      {...props}
      d='M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z'
    />
  )
}
