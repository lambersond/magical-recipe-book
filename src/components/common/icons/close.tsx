import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function CloseIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='CloseIcon'
      {...props}
      d='M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z'
    />
  )
}
