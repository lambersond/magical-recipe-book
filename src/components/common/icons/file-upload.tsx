import { Icon } from './icon'
import type { SVGAttributes } from 'react'

export function FileUploadIcon(props: Readonly<SVGAttributes<SVGElement>>) {
  return (
    <Icon
      data-testid='FileUploadIcon'
      {...props}
      d='M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z'
    />
  )
}
