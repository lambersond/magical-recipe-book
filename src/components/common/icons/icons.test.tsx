import { render } from '@test-utils'
import { AccountIcon } from './account'
import { AddIcon } from './add'
import { ArrowDownIcon } from './arrow-down'
import { CheckIcon } from './check'
import { CheckboxIcon } from './checkbox'
import { CheckboxBlankIcon } from './checkbox-blank'
import { ChevronLeftIcon } from './chevron-left'
import { ChevronRightIcon } from './chevron-right'
import { ClockIcon } from './clock'
import { CloseIcon } from './close'
import { CogIcon } from './cog'
import { DeleteIcon } from './delete'
import { EditIcon } from './edit'
import { FileUploadIcon } from './file-upload'
import { LaunchIcon } from './launch'
import { LoginIcon } from './login'
import { LogoutIcon } from './logout'
import { MailIcon } from './mail'
import { MoreVertIcon } from './more-vert'
import { NewIcon } from './new'
import { SaveIcon } from './save'

describe('components/common/icons', () => {
  it.each([
    ['AccountIcon', AccountIcon],
    ['AddIcon', AddIcon],
    ['ArrowDownIcon', ArrowDownIcon],
    ['CheckIcon', CheckIcon],
    ['CheckboxBlankIcon', CheckboxBlankIcon],
    ['CheckboxIcon', CheckboxIcon],
    ['ChevronLeftIcon', ChevronLeftIcon],
    ['ChevronRightIcon', ChevronRightIcon],
    ['ClockIcon', ClockIcon],
    ['CloseIcon', CloseIcon],
    ['CogIcon', CogIcon],
    ['DeleteIcon', DeleteIcon],
    ['EditIcon', EditIcon],
    ['FileUploadIcon', FileUploadIcon],
    ['LaunchIcon', LaunchIcon],
    ['LoginIcon', LoginIcon],
    ['LogoutIcon', LogoutIcon],
    ['MailIcon', MailIcon],
    ['MoreVertIcon', MoreVertIcon],
    ['NewIcon', NewIcon],
    ['SaveIcon', SaveIcon],
  ])('should match snapshot - %s', (_name, Component) => {
    const { asFragment } = render(<Component />)

    expect(asFragment()).toMatchSnapshot()
  })
})
