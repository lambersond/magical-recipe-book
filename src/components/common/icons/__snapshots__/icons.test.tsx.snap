// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/common/icons should match snapshot - AccountIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="AccountIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - AddIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="AddIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - ArrowDownIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="ArrowDownIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M5.23 7.21a.75.75 0 011.06.02L10 10.94l3.71-3.71a.75.75 0 111.06 1.06l-4.24 4.24a.75.75 0 01-1.06 0L5.23 8.27a.75.75 0 01.02-1.06z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - CheckIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="CheckIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - CheckboxBlankIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="CheckboxBlankIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - CheckboxIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="CheckboxIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 16H5V5h14zM17.99 9l-1.41-1.42-6.59 6.59-2.58-2.57-1.42 1.41 4 3.99z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - ChevronLeftIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="ChevronLeftIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - ChevronRightIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="ChevronRightIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - ClockIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="ClockIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - CloseIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="CloseIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - CogIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="CogIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - DeleteIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="DeleteIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - EditIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="EditIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.996.996 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - FileUploadIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="FileUploadIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M9 16h6v-6h4l-7-7-7 7h4zm-4 2h14v2H5z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - LaunchIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="LaunchIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - LoginIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="LoginIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - LogoutIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="LogoutIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - MailIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="MailIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - MoreVertIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="MoreVertIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - NewIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="NewIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="m19 9 1.25-2.75L23 5l-2.75-1.25L19 1l-1.25 2.75L15 5l2.75 1.25zm0 6-1.25 2.75L15 19l2.75 1.25L19 23l1.25-2.75L23 19l-2.75-1.25zm-7.5-5.5L9 4 6.5 9.5 1 12l5.5 2.5L9 20l2.5-5.5L17 12zm-1.51 3.49L9 15.17l-.99-2.18L5.83 12l2.18-.99L9 8.83l.99 2.18 2.18.99z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`components/common/icons should match snapshot - SaveIcon 1`] = `
<DocumentFragment>
  <svg
    data-testid="SaveIcon"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3m3-10H5V5h10z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;
